DREAM3DReviewFilters/SurfaceMeshToWaveFront.cpp

```diff
@@ -0,0 +1,305 @@
+  clearWarningCode();
+  setCancel(false);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void SurfaceMeshToWaveFront::setupFilterParameters()
+{
+  FilterParameterVectorType parameters;
+
+  parameters.push_back(SIMPL_NEW_OUTPUT_FILE_FP("Output Wavefront File", OutputWaveFrontFile, FilterParameter::Parameter, SurfaceMeshToWaveFront, "*.obj", "Wavefront Object File"));
+
+  DataArraySelectionFilterParameter::RequirementType req;
+  req.dcGeometryTypes = IGeometry::Types(1, IGeometry::Type::Triangle);
+  req.amTypes = AttributeMatrix::Types(1, AttributeMatrix::Type::Vertex);
+  req.daTypes = QVector<QString>(1, SIMPL::TypeNames::Int8);
+  std::vector<std::vector<size_t>> comp;
+  comp.push_back(std::vector<size_t>(1, 1));
+  req.componentDimensions = comp;
+  parameters.push_back(SIMPL_NEW_DA_SELECTION_FP("Surface Mesh Node Type", SurfaceMeshNodeTypeArrayPath, FilterParameter::RequiredArray, SurfaceMeshToWaveFront, req));
+
+  setFilterParameters(parameters);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void SurfaceMeshToWaveFront::dataCheck()
+{
+  clearErrorCode();
+  clearWarningCode();
+
+  FileSystemPathHelper::CheckOutputFile(this, "Output Wavefront File", getOutputWaveFrontFile(), true);
+
+  std::vector<size_t> dims = {1};
+  m_SurfaceMeshNodeTypePtr = getDataContainerArray()->getPrereqArrayFromPath<Int8ArrayType>(this, getSurfaceMeshNodeTypeArrayPath(), dims);
+  if(nullptr != m_SurfaceMeshNodeTypePtr.lock())
+  {
+    m_SurfaceMeshNodeType = m_SurfaceMeshNodeTypePtr.lock()->getPointer(0);
+  }
+
+  DataContainer::Pointer sm = getDataContainerArray()->getPrereqDataContainer(this, getSurfaceMeshNodeTypeArrayPath().getDataContainerName(), false);
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  TriangleGeom::Pointer triangleGeom = sm->getPrereqGeometry<TriangleGeom>(this);
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void SurfaceMeshToWaveFront::execute()
+{
+  initialize();
+  dataCheck();
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  DataContainer::Pointer dc = getDataContainerArray()->getDataContainer(getSurfaceMeshNodeTypeArrayPath().getDataContainerName());
+  TriangleGeom::Pointer triangleGeom = dc->getGeometryAs<TriangleGeom>();
+  MeshIndexType numberOfTriangles = triangleGeom->getNumberOfTris();
+  SharedTriList::Pointer triangles = triangleGeom->getTriangles();
+  SharedVertexList::Pointer vertices = triangleGeom->getVertices();
+  size_t numberOfVertices = triangleGeom->getNumberOfVertices();
+
+  std::ofstream ss;
+  ss.open(getOutputWaveFrontFile().toStdString().c_str());
```

Comment:
```suggestion
  std::ofstream file(getOutputWaveFrontFile().toStdString().c_str());
```
Can combine the open call into the constructor. Rename std::ofstream variable to prevent shadowing of QString below.